include/ginkgo/core/solver/batch_solver.hpp

```diff
@@ -107,6 +109,21 @@ class EnableBatchSolver
         return right_scaling_;
     }
 
+    /**
+     * Get the residual tolerance used by the solver.
+     *
+     * @return The residual tolerance.
+     */
+    double get_residual_tolerance() const { return residual_tol_; }
```

Comment:
I think we should maintain just one copy of the tolerance instead of two separate copies. If the users sets a tolerance, then we just modify the existing one instead of adding one more parameter. the `parameters_` struct already has a getter as well

Reply:
Alternatively: Get rid of the parameters member entirely. It's hidden in one of the factory macros, but TBH I dislike how well they hide the additional member they add. I worked entirely without the macros in #1082, and it felt much cleaner :)

Reply:
There are two types of parameters: Ones that describe how to generate an object, but can be discarded afterwards, and ones that describe the runtime behavior of the object. IMO, we should only keep the latter inside the generated object.

Reply:
@upsj That has the advantage that I could add the residual tolerance, getter and setter only to the base class. The problem with the factory macros is that the entire struct needs to be in the derived class, so we'd need to duplicate the getter and setter on every batched iterative solver, which should ideally be unnecessary.

Reply:
But I think having two things settable in two different ways might bring a lot more confusion, so we should unify those.